swarm/network/stream/delivery.go

```diff
@@ -173,12 +174,48 @@ func (d *Delivery) Close() {
 
 // FindPeer is returning the closest peer from Kademlia that a chunk
 // request hasn't already been sent to
-func (d *Delivery) FindPeer(req *storage.Request) (*Peer, error) {
+func (d *Delivery) FindPeer(ctx context.Context, req *storage.Request) (*Peer, error) {
 	var sp *Peer
 	var err error
 
+	osp, _ := ctx.Value("remote.fetchh").(opentracing.Span)
+
 	depth := d.kad.NeighbourhoodDepth()
 
+	originPo := -1
+	d.kad.EachConn(req.Addr[:], 255, func(p *network.Peer, po int) bool {
```

Comment:
just use `chunk.Proximity` here to get originPO

Reply:
@zelig, right of course, we have access to the baseAddr and to req.Addr so it is literally just a function call :)

Reply:
@zelig `origin` could be empty (if the request comes from our node through the `fileStore`), or it could be the enode of the peer that requests that chunk from us. Therefore we need to be able to convert the `enode` to an overlay address and get the chunk.Proximity... right now this is extracted into its own method and works, since we iterate Kademlia. I am not sure if we have functionality to directly get an overlay address based on an enode.